### PR TITLE
Refactor configuration.ts

### PR DESCRIPTION
**Pull request by quinn-dev**
**Changes:**
- Completely refactored `configuration.ts`
- Added validation of `config.json`
- Add colored errors using `chalk`
- Fixed minor bug: When the prompts dialog was aborted (for example using `Strg` + `C`), the tool created an invalid `config.json`

**New dependencies:**
- [superstruct](https://www.npmjs.com/package/superstruct) (for object validation)
- [chalk](https://www.npmjs.com/package/chalk) (for colored error output)
- [is-url](https://www.npmjs.com/package/is-url) (for URL validation)

### CODE REVIEW

File: src/configuration.ts [hunk 1]
```diff
@@ -2,7 +2,18 @@ import * as fs from "fs";
...
 import Debug from "debug";
+import {
+  define,
+  assert,
+  optional,
+  object,
+  array,
+  string,
+  number,
+} from "superstruct";
```
Comment (marcoklein) [inline src/configuration.ts:13]: nice solution with superstruct, wasn't aware of that lib

File: src/configuration.ts [hunk 2]
```diff
@@ -12,90 +23,158 @@ export interface Config {
...
-      if (!anotherUrl) {
-        break;
+    configuration = await generateNewConfig();
+  }
+
+  try {
+    assert(configuration, Config);
+  } catch (e) {
+    debug(error("Invalid configuration:", e));
+    if (await askToCreateNewConfig) {
```
Comment (marcoklein) [inline src/configuration.ts:165]: I think this should call the function with `askToCreateNewConfig()`?

File: yarn.lock [hunk 3]
```diff
@@ -2053,6 +2070,11 @@ kind-of@^6.0.3:
   resolved "https://registry.yarnpkg.com/kind-of/-/kind-of-6.0.3.tgz#07c05034a6c349fa06e24fa35aa76db4580ce4dd"
   integrity sha512-dcS1ul+9tmeD95T+x28/ehLgd9mENa3LsvDTtzm3vyBEO7RPptvAD+t44WVXaUjTBRcrpFeFlC8WCruUR456hw==
 
+kleur@^3.0.3:
```
Comment (marcoklein) [inline yarn.lock:2073]: Just to point out: chalk could be replaced by kleur cause kleur is a peer dependency of prompts. But the latest version has the major version 4 so it's fine to use chalk.

Another coloring option would be the one from debug https://www.npmjs.com/package/debug#namespace-colors.

No need to update though, chalk is doing the job.

File: src/index.ts [hunk 4]
```diff
@@ -8,6 +8,9 @@ import { checkForUrlWithCode } from "./zentrum";
 import { tmpdir } from "os";
 import * as path from "path";
 const debug = Debug("impftermin:main");
+import { bgRedBright, whiteBright } from "chalk";
+
+export const error = (...text: unknown[]) => bgRedBright(whiteBright(...text));
```
Comment (marcoklein) [inline src/index.ts:13]: Could you give this a more descriptive name like `colorErrorText` or something?

**Review by marcoklein — CHANGES_REQUESTED** [review]
Nice implementation - please see my comments.

File: src/configuration.ts (hunk 2 above)
Comment (quinn-dev) [reply src/configuration.ts:165]: Yes, of course. I'm surprised I didn't realize this during testing.

File: src/index.ts (hunk 4 above)
Comment (quinn-dev) [reply src/index.ts:13]: How about `coloredError`?

**Comment by quinn-dev** [comment]
I implemented the changes in #00b4852.

**Review by marcoklein — APPROVED** [review]

**Pull request merged**